lib/features/user/domain/usecases/update_user_details.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:coffeecard/core/errors/failures.dart';
+import 'package:coffeecard/core/usecases/usecase.dart';
+import 'package:coffeecard/features/user/data/datasources/user_remote_data_source.dart';
+import 'package:coffeecard/features/user/domain/entities/user.dart';
+import 'package:coffeecard/models/account/update_user.dart';
+import 'package:dartz/dartz.dart';
+import 'package:equatable/equatable.dart';
+
+class UpdateUserDetails implements UseCase<User, Params> {
+  final UserRemoteDataSource dataSource;
+
+  UpdateUserDetails({required this.dataSource});
+
+  @override
+  Future<Either<Failure, User>> call(Params params) async {
+    return dataSource.updateUserDetails(
+      UpdateUser(
+        name: params.name,
+        email: params.email,
+        encodedPasscode: params.encodedPasscode,
+        privacyActivated: params.privacyActivated,
+        occupationId: params.occupationId,
+      ),
+    );
+  }
+}
+
+class Params extends Equatable {
+  final String? name;
```

Comment:
Name, email, and passcode appears non-nullable on the backend. Hve you tested this works?

Reply:
you would only need to supply the details you want to update as was also the case before. The logic is the same

Reply:
What I was trying to get at is that on the backend side of things, the updateUserRequest is declared as follows:
`       public string Name { get; set; }
        
        public string Email { get; set; }
        
        public bool? PrivacyActivated { get; set; }
        
        public int? ProgrammeId { get; set; }
        
        public string Password { get; set; }
`

Though it seems that due to them not being marked as required it somehow stille works when given null values for the non-nullable fields. Dotnet is weird sometimes 🤔